turbomoleio/core/utils.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# The turbomoleio package, a python interface to Turbomole
+# for preparing inputs, parsing outputs and other related tools.
+#
+# Copyright (C) 2018-2021 BASF SE, Matgenix SRL.
+#
+# This file is part of turbomoleio.
+#
+# Turbomoleio is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Turbomoleio is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with turbomoleio (see ~turbomoleio/COPYING). If not,
+# see <https://www.gnu.org/licenses/>.
+
+"""
+Module containing miscellaneous utilities
+"""
+
+from monty.tempfile import ScratchDir
+import re
+import pexpect.popen_spawn
+import signal
+
+
```

Comment:
why not having these as methods in DefineRunner?

Maybe `get_tm_version` could be a bit of a stretch, since it also runs in a temporary directory and could stay out as an external function, but moving `define_quit` would allow to limit the use of pexpect to the define module. What do you think?

Reply:
It's indeed a possibility. I thought I would put it outside because 1/ it's not really using define and 2/ DefineRunner would/should be able to execute define for the current (last) version of Turbomole, while the define_quit is just to get the version (which might be an earlier version). I am ok moving the define_quit to the DefineRunner if you think it is better.

Reply:
I see. To me it would seem cleaner to have it in DefineRunner and I would not expect to change the fact that it can quit with `qq`. However I do not have such a strong opinion on this.

Reply:
Well also I want to make sure that there is no cyclic import, e.g. suppose the definerunner needs to know which version of TM is used (even though we are going towards a "last tmio version supports last tm", so it should in principle stay like that), then its using get_tm_version which is in turn using definerunner. I would probably keep it this way.